app/api/emails.py

```
@@ -0,0 +1,182 @@
+    Get email status by ID
+    
+    Returns the current status, retry count, and other details about the email.
+    """
+    try:
+        email = await email_service.get_by_id(db, email_id)
+        
+        if not email or email.caller_id != authenticated_caller_id:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail=f"Email {email_id} not found"
+            )
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)